website/content/docs/waypoint-hcl/variables/input.mdx

```diff
@@ -0,0 +1,353 @@
+---
+layout: docs
+page_title: Input Variables - Variables - waypoint.hcl
+description: |-
+  Input variables allow you to customize Waypoint configuration according to 
+  set parameters. This section documents input variable syntax, including how to
+  define, reference, and use input variables.
+---
+
+# Input Variables
+
+Input variables let you customize aspects of Waypoint configuration with values
+that may change, or that need to be set per Waypoint run.
+
+You can parameterize your configuration by declaring variables in the
+waypoint.hcl, and setting their values using CLI options, environment variables,
+`auto` files, and the Waypoint server UI.
+
+> **Hands-on:** Try the [Input Variables]() tutorial on HashiCorp Learn.
+
+## Declaring an Input Variable
+
+Each input variable referenced in the waypoint.hcl must be declared using a
+`variable` block:
```

Comment:
I would link to the stanza reference documentation for variable here. I don't see that in this PR so I recommend adding that, example: https://www.waypointproject.io/docs/waypoint-hcl/app That should be the exhaustive reference of the syntax and available args. You can keep everything here, but that should exist too!

The idea there is anyone should be able to look at that sidebar and see every valid stanza in a waypoint file.